app.go

```diff
@@ -914,7 +873,7 @@ func (a *Application) Commit() abci.ResponseCommit {
 
 	// Update the last block header before emitting events in case the subscribers attempt to access
 	// the latest committed state as soon as they receive an event.
-	a.lastBlockHeader = a.curBlockHeader
+	atomic.StorePointer(&a.lastBlockHeader, unsafe.Pointer(&a.curBlockHeader))
```

Comment:
This doesn't look right, I'm pretty sure `&a.curBlockHeader` is going to remain the same, meaning the last block header will always reference the current block header.

Reply:
good catch